### DISCUSSION TITLE
Self hosted collaboration server?

### DISCUSSION THREAD

**Discussion by nilskch** [zed-industries/zed#13503] [Feature Requests]
Is is possible or on the roadmap to self-host the collaboration server? My empoyer does not want to allow access to `collab.zed.dev` due to to governance issues.

**Comment by skewballfox** [comment]
So I have some prior knowledge about federal requirements for accessing protected source code that may be beneficial to consider:
1. Often connections can only be made over a private subnet (so self-hosted server or direct ssh is required)
2. Data accessed *cannot* exist in an unencryped format on disk or in the filesystem ([relevant SO question](https://stackoverflow.com/questions/66286406/does-vscode-remote-ssh-extension-in-remote-development-store-any-info-client-sid)). With linux you could probably do this with an encrypted tmpfs

If you meet these requirements, you'll be the only game in town. IntelliJ and Sundry don't meet these requirements, nor does vscode(without the interventions mentioned in the SO post which breaks updating). I'm 95% sure sublime doesn't. If there is one place a dual licence or premium feature model could come in handy, it's probably with meeting those two requirements.

**Comment by uhryniuk** [comment]
> Commenting so this issue gets more visibility.

Would love this feature :100:

**Comment by bmoore** [comment]
👍 also wondering the same thing

**Comment by KaKi87** [comment]
[Yes](https://github.com/zed-industries/zed/issues/8260#issuecomment-1965463519).

**Comment by Simx72** [comment]
Was about to post the exact same thing

**Comment by kabakaev** [comment]
Isn't it published at https://github.com/zed-industries/zed/tree/main/crates/collab already?

**Reply by zeigerpuppy** [reply]
from what I can see there, the sync server is available (good), but is reliant on the Zed authentication, so not reall a local-only solution without Auth being handled locally too.

**Reply by zeigerpuppy** [reply]
see https://github.com/zed-industries/zed/blob/main/docs/src/development/local-collaboration.md#running-a-local-collab-server

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
